src/Client.h

```diff
@@ -140,21 +140,53 @@ class Client : public ClientBase{
 		int serverNo = 0;
 
 		TicksTime rxTime;
-
-		ret = msg_recvfrom(ifd,
-				           g_fds_array[ifd]->recv.cur_addr + g_fds_array[ifd]->recv.cur_offset,
-				           g_fds_array[ifd]->recv.cur_size,
-				           &recvfrom_addr);
-		if (ret == RET_SOCKET_SHUTDOWN) {
-			if (g_fds_array[ifd]->sock_type == SOCK_STREAM) {
-				exit_with_log("A connection was forcibly closed by a peer",SOCKPERF_ERR_SOCKET,g_fds_array[ifd]);
+#ifdef  USING_VMA_EXTRA_API
+		if (!g_vma_api || g_pApp->m_const_params.fd_handler_type != VMAPOLL)
+#endif
```

Comment:
Unfortunately no this is the best that we can provide and it will not work otherwise.